### ISSUE TITLE
The resolve() call is missing in the stop function. (IOS)

### ISSUE THREAD

**Issue by KovendiMozes** [Cap-go/native-audio#131]
https://github.com/Cap-go/native-audio/blob/e9b24a43cb9b66eded64c0550643930866e80c1d/ios/Plugin/Plugin.swift#L229
	
The linked code is missing the resolve() call and because of this I cannot wait for the promise to resolve in this function.

Could someone possibly fix this? Thank you in advance.

**Comment by riderx** [comment]
You can PR it, i merge and deploy direct:)

**Comment by KovendiMozes** [comment]
I got stuck at the point where I don't have permission to push the branch I created.

![Screenshot from 2024-09-06 12-50-06](https://github.com/user-attachments/assets/08f01593-ae7e-436f-8d9d-53196528f7cf)

If you'd rather do it yourself, here are the changes I need.

![Screenshot from 2024-09-06 13-11-58](https://github.com/user-attachments/assets/bc79418c-e8da-4105-806c-a42f4364d1ec)

In my opinion:
Meanwhile, I noticed that a few more places are missing this 'call' function, which would be good for added security.

iOS
Reject:
1. if the condition "if" is not met
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/ios/Plugin/Plugin.swift#L138
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/ios/Plugin/Plugin.swift#L142
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/ios/Plugin/Plugin.swift#L145

2. before return
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/ios/Plugin/Plugin.swift#L187
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/ios/Plugin/Plugin.swift#L197
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/ios/Plugin/Plugin.swift#L207
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/ios/Plugin/Plugin.swift#L216
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/ios/Plugin/Plugin.swift#L237
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/ios/Plugin/Plugin.swift#L262
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/ios/Plugin/Plugin.swift#L273
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/ios/Plugin/Plugin.swift#L283

Android
Resolve:
https://github.com/Cap-go/native-audio/blob/ec55599d65a29585d9b03ed5a829d0b9918926d9/android/src/main/java/ee/forgr/audio/NativeAudio.java#L396

**Comment by riderx** [comment]
Yes to push it was required to fork first, github is a bit dumb for that. We will look at it thanks

**Comment by riderx** [comment]
@KovendiMozes I did some fix:
1.if the condition "if" is not met:
- this one is correct, I changed the code to make it more clear
- correct as well
- incorrect the code throw and it's catched with and do call reject
2. all related to getAudioAsset function are reject in the function so only return is normal
3.
- correct

I did also simplify the code to remove a lot of intrication of if, and put the error in IF instead of good cases.
